File: CPP/No413.cpp
/**
 * Created by Xiaozhong on 2020/9/16.
 * Copyright (c) 2020/9/16 Xiaozhong. All rights reserved.
 */

#include "iostream"
#include "vector"

using namespace std;

class Solution {
public:
    int numberOfArithmeticSlices(vector<int> &A) {
        vector<int> dp(A.size(), 0);
        int sum = 0;
        for (int i = 2; i < A.size(); ++i) {
            if (A[i] - A[i - 1] == A[i - 1] - A[i - 2]) {
                dp[i] = 1 + dp[i - 1];
                sum += dp[i];
            }
        }
        return sum;
    }
};

int main() {
    Solution s;
    vector<int> nums = {1, 2, 3, 4};
    cout << s.numberOfArithmeticSlices(nums);
}

File: Java/src/No413.java
/*
 * Copyright (c) 8/10/2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

public class No413 {
    public int numberOfArithmeticSlices(int[] nums) {
        int cnt = 0, ans = 0;
        for (int i = 2; i < nums.length; ++i) {
            if (nums[i] - nums[i - 1] == nums[i - 1] - nums[i - 2]) ans += ++cnt;
            else cnt = 0;
        }
        return ans;
    }
}
